### PR TITLE
Add wpt tests for send

### LINKED ISSUE #9357: Add tests for exercising XHR.send(data) method
New tests should live under `tests/wpt/mozilla/tests/mozilla`. We'll need to feed in the `.send()` method with different types of data that we currently support, namely `USVString`s, `URLSearchParams` and `Blob`s.

Spec: https://xhr.spec.whatwg.org/#the-send%28%29-method

### PR DESCRIPTION
**Pull request by dlrobertson**
Add wpt tests for send that do not set the responseType and compare the result against a string.

Fixes #9357 

r? @KiChjang

---

This change is [Review on Reviewable](https://reviewable.io/reviews/servo/servo/9971)

### CODE REVIEW

**Comment by danlrobertson** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/servo/servo/9971#-KCbk6FvVnH6KTJN049E:-KCbk6FwDr3b686dRixa:1964149046) (no related file):*
I was confused as to what tests should be added, so I based it off of the comment by @jdm. If you would like tests for `URLSearchParams` etc, just let me know

---

*[tests/wpt/mozilla/meta/MANIFEST.json, line 6110 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9971#-KCbj4Yx5gva8HScMFuL:-KCbj4YyXtjXpsjkc9UA:1359212003) ([raw file](https://github.com/servo/servo/blob/c6093f3a31a03e5add03ac74d05c44e301c3b068/tests/wpt/mozilla/meta/MANIFEST.json#L6110)):*
These were the only two tests among `tests/wpt/web-platform-tests/XMLHttpRequest/send-data-*` that either set the responseType (`blob`) or would fail without comparing the data against a string (`arraybuffer`).

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9971)*

**Comment by KiChjang** [comment]
We're not supposed to add in WPT tests, per the discussion in the issue. Instead, we need unit tests, and they're written in Rust.

**Comment by jdm** [comment]
No, this PR is doing what was requested in the original issue - we're adding servo-specific WPT tests that don't get upstreamed that allow us to test the things we care about without relying on features we don't implement yet.

**Comment by danlrobertson** [comment]
So what should I do about `resources/content.py`? the lint requests a license, but the file is copied from `web-platform-tests`.

If licensing is a problem, I can always just make a new `content.py` and add our license header.

File: tests/wpt/mozilla/meta/MANIFEST.json [hunk 1]
```diff
@@ -6096,6 +6096,18 @@
...
+            "path": "mozilla/send-without-content-type-arraybuffer.htm",
+            "url": "/_mozilla/mozilla/send-without-content-type-arraybuffer.htm"
+            }
+        ],
+        "mozilla/send-without-content-type-blob.htm": [
+          {
+            "path": "mozilla/send-without-content-type-blob.htm",
+            "url": "/_mozilla/mozilla/send-without-content-type-blob.htm"
+          }
+        ],
```
Comment (KiChjang) [inline tests/wpt/mozilla/meta/MANIFEST.json]: Is this manually typed? Tidy doesn't like the indentation of the brackets.

Comment (danlrobertson) [inline tests/wpt/mozilla/meta/MANIFEST.json]: Good point! Yes it was, updated using `mach create-wpt`

**Comment by KiChjang** [comment]
Just one small nit - why are the test files named *-without-content-type-*?

**Comment by danlrobertson** [comment]
Ah! Sorry, should have been send-without-response-type-*. Let me know if you prefer another name.

---

Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9971#-:-KD-EmBSldzedQcDykev:-1011028422)*

**Comment by KiChjang** [comment]
This still is a misnomer - the response being parsed is a text response. I suggest removing the -without-response-type altogether since this isn't relevant to the test that you have wrote.

**Comment by danlrobertson** [comment]
True, changed it to just `send-*`.

File: tests/wpt/mozilla/tests/mozilla/send-arraybuffer.htm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                        assert_equals(xhr.status, 200);
+                        assert_equals(xhr.response, "[object ArrayBuffer]");
+
+                        test.done();
+                    });
+                }
+            };
+
+            xhr.open("POST", "./resources/content.py", true);
+            xhr.responseType = "text";
```
Comment (KiChjang) [inline tests/wpt/mozilla/tests/mozilla/send-arraybuffer.htm]: We shouldn't be explicitly setting the responseType here - we should instead assert that the response we read from xhr is a text string.

**Comment by KiChjang** [comment]
I'm unsure about whether we should include tests for `URLSearchParams`. If we do include it, I think it should be a WPT test instead of a mozilla test.

**Comment by danlrobertson** [comment]
> I'm unsure about whether we should include tests for URLSearchParams. If we do include it, I think it should be a WPT test instead of a mozilla test.

Makes sense!

I removed the `responseType = "text"`.

**Comment by KiChjang** [comment]
I'm going to go ahead and merge this. Filed w3c/web-platform-tests#2709 for the missing test coverage. Thanks for working on this!

@bors-servo r+

**Comment by Ms2ger** [comment]
@bors-servo retry
* #10071

**Pull request merged**